### PR TITLE
The `print()` function didn't print

### PR DESCRIPTION
**Pull request by softdevca**
The `print(ast`) function didn't print anything, it merely formatted the `Ast` as a string. This was confusing.

To make it more clear the function has been replaced by an implementation of `Display` for `Ast` that has the same formatting behavior.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -7,6 +7,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 ## [1.0.0-rc.5] - 2024-07-06
 ### Changed
 - **[BREAKING CHANGE]** The `types` module is now named `luau`.
+- **[BREAKING CHANGE]** The `print()` function has been removed, use `ast.to_string()` instead.
```
Comment (Kampfkarren) [inline CHANGELOG.md:10]: Put this in a new "Unreleased" section with no date

Comment (softdevca) [reply CHANGELOG.md]: Done.

**Comment by Kampfkarren** [comment]
Apologies for delays :) Looks like a trivial `cargo fmt` is all this needs now

**Comment by softdevca** [comment]
> Apologies for delays :) Looks like a trivial `cargo fmt` is all this needs now

Done, thanks!

**Review by DataM0del — APPROVED** [review]
Other than the clippy errors (which can be fixed using PR #316), this PR makes sense.
Although it does break compatibility.

**Pull request merged**